frontend/src/components/Header/HeaderAuth.tsx

```diff
@@ -1,19 +1,34 @@
+import { useCookies } from 'react-cookie';
+import { Link } from 'react-router-dom';
+
 const HeaderAuth = () => {
-  return (
-    <div className="hidden md:flex items-center justify-end md:flex-1 lg:w-0">
-      <a
-        href="/signin"
-        className="whitespace-nowrap text-base font-medium text-gray-500 hover:text-gray-900"
-      >
-        Sign in
-      </a>
-      <a
-        href="/signup"
-        className="ml-8 whitespace-nowrap inline-flex items-center justify-center px-4 py-2 border border-transparent rounded-md shadow-sm text-base font-medium text-white bg-indigo-600 hover:bg-indigo-700"
-      >
-        Sign up
-      </a>
-    </div>
+  const [cookies, setCookies] = useCookies();
+
+  return !cookies.user ? (
+    <>
+      <div className="hidden md:flex items-center justify-end md:flex-1 lg:w-0">
+        <Link
+          to="/signin"
+          className="whitespace-nowrap text-base font-medium text-gray-500 hover:text-gray-900"
+        >
+          Sign in
+        </Link>
+        <Link
+          to="/signup"
+          className="ml-8 whitespace-nowrap inline-flex items-center justify-center px-4 py-2 border border-transparent rounded-md shadow-sm text-base font-medium text-white bg-indigo-600 hover:bg-indigo-700"
+        >
+          Sign up
+        </Link>
+      </div>
+    </>
+  ) : (
+    <>
+      <div className="hidden md:flex items-center justify-end md:flex-1 lg:w-0">
+        <p className="whitespace-nowrap text-base font-medium text-gray-500 hover:text-gray-900">
+          Welcome Back
+        </p>
+      </div>
+    </>
```

Comment:
`<></>` fragments are useful when you have 2 or more child element to wrap, since we have only 1 parent element, we can still use this as a wrapper
```suggestion
      <div className="hidden md:flex items-center justify-end md:flex-1 lg:w-0">
        <p className="whitespace-nowrap text-base font-medium text-gray-500 hover:text-gray-900">
          Welcome Back
        </p>
      </div>
```

Reply:
okay, i didnt necessarily know what was the use of the <></>, the more u know

Reply:
updated